Adaq4224 scale fix

machschmitt opened the pull request:
Update ad4630 driver with fixes for ADAQ4224 scale.

nunojsa commented on drivers/iio/adc/ad4630.c:
```diff
@@ -443,9 +461,11 @@ static void ad4630_fill_scale_tbl(struct ad4630_state *st)
-		val = DIV_ROUND_CLOSEST(val * 1000, ad4630_gains[i]);
-
-		tmp2 = shift_right((u64)val * NANO, val2);
+		/* Multiply by MILLI here to avoid losing precision */
```
I would remove this comment...

nunojsa commented on drivers/iio/adc/ad4630.c:
```diff
@@ -141,13 +141,31 @@ enum {
+	AD4630_222_GAIN = 2,
+	AD4630_667_GAIN = 3,
+	AD4630_MAX_PGA,
+};
```
This patch looks more like an improvement rather than a fix... Not sure i'm convinced about the tag

machschmitt replied:
Ideally, the scale would be good enough since the patch introducing support for these parts. Also, the whole point of these patches is to get the driver to a state where users would be satisfied with the readings from the ADC. No strong opinion about the tag though. I will probably keep the tag, but will remove it if you think it's better not to have that.

nunojsa replied:
> No strong opinion about the tag though. I will probably keep the tag, but will remove it if you think it's better not to have that.

The question is... Is this so fundamental and important that we should backport it? I'm still not convinced about the tag but no strong opinion as well. But keep in mind that if you keep the tag, then let's treat it as a fix and backport it to 2022_R2 (if these devices are supported in there)

nunojsa commented on drivers/iio/adc/ad4630.c:
```diff
@@ -1481,6 +1476,11 @@ static int ad4630_probe(struct spi_device *spi)
+	if (st->pga_gpios) {
+		ad4630_fill_scale_tbl(st);
+		ad4630_set_pga_gain(indio_dev, 0);
+	}
```
Seems like an unrelated change... Why do we have it? If it's a needed move (makes more sense or so) please mention it in the commit message so me and others don't have to wonder about it :)

nunojsa replied:
Ahhh I see. No need to re-spin only because of it but if you do, please include that in the commit message.

nunojsa commented on drivers/iio/adc/ad4630.c:
```diff
@@ -144,7 +144,7 @@ enum {
  * Gains computed as fractions of 1000 so they can be expressed by integers.
  */
-static const int ad4630_gains[4] = {
+static const int ad4630_gains[] = {
```
unrelated change...

nunojsa commented on drivers/iio/adc/ad4630.c:
```diff
@@ -435,15 +438,15 @@ static int ad4630_set_chan_offset(struct iio_dev *indio_dev, int ch, int offset)
-		tmp2 = shift_right((s64)val * 1000000000LL, val2);
-		tmp0 = (int)div_s64_rem(tmp2, 1000000000LL, &tmp1);
+		tmp2 = shift_right((u64)val * NANO, val2);
+		tmp0 = (int)div_s64_rem(tmp2, NANO, &tmp1);
```
Unrelated changes... mention it in the commit. Since you're doing these small changes, I think you can also get rid of the (int) cast.

nunojsa commented on drivers/iio/adc/ad4630.c:
```diff
@@ -452,17 +455,19 @@ static void ad4630_fill_scale_tbl(struct ad4630_state *st)
-	gain_nano = DIV_ROUND_CLOSEST_ULL(gain_nano, (vref / 1000000) * 2);
-	gain_idx = find_closest(gain_nano, ad4630_gains, ARRAY_SIZE(ad4630_gains));
+	tmp = gain_nano << precision;
+	tmp = DIV_ROUND_CLOSEST_ULL(tmp, NANO);
```
one liner?

`DIV_ROUND_CLOSEST_ULL(gain_nano << precision, NANO)`

machschmitt replied:
respining to clean up unrelated changes and improve commit messages

nunojsa reviewed (commented):
Small nits otherwise LGTM.

machschmitt commented:
I don't think adaq4224 is supported in 2022_R2 but removed the fixes tag from the second patch anyway. It is not critical and I can't really convince myself of that tag either.
Removed unrelated changes.
Took online expression for part of the math.
Added comment about move of gpio setup in commit description.
I would like to keep the comments. Getting the scale correct took a bit of tinkering. I think some comments might help if ever need to look into this again.

thanks Nuno.
